MSGraphSDKNXOAuth2/Auth/NXOAuth2AuthenticationProvider.m

```diff
@@ -188,9 +188,17 @@ - (void)loginWithViewController:(UIViewController*)viewController
     }];
 }
 
-//- (BOOL)loginSilent {
-//    // Placeholder: Load an existing account from the account store
-//}
+- (BOOL)loginSilent {
+    // Placeholder: Load an existing account from the account store
+    int accountCount = [[NXOAuth2AccountStore sharedStore] accounts].count;
+    
+    if (accountCount == 0) {
+        return NO;
+    } else {
+        self.userAccount = [[NXOAuth2AccountStore sharedStore] accounts][accountCount - 1];
```

Comment:
Sounds good to me.

From:  Jason Kim <notifications@github.com>
Reply-To:  microsoftgraph/msgraph-sdk-ios-nxoauth2-adapter <reply@reply.github.com>
Date:  Tuesday, May 17, 2016 at 3:19 PM
To:  microsoftgraph/msgraph-sdk-ios-nxoauth2-adapter <msgraph-sdk-ios-nxoauth2-adapter@noreply.github.com>
Cc:  Kevin Lam <klam@outlook.com>, Comment <comment@noreply.github.com>
Subject:  Re: [microsoftgraph/msgraph-sdk-ios-nxoauth2-adapter] add login silent implementation (#2)


In MSGraphSDKNXOAuth2/Auth/NXOAuth2AuthenticationProvider.m:
> @@ -188,9 +188,17 @@ - (void)loginWithViewController:(UIViewController*)viewController
>      }];
>  }
>  
> -//- (BOOL)loginSilent {
> -//    // Placeholder: Load an existing account from the account store
> -//}
> +- (BOOL)loginSilent {
> +    // Placeholder: Load an existing account from the account store
> +    int accountCount = [[NXOAuth2AccountStore sharedStore] accounts].count;
> +    
> +    if (accountCount == 0) {
> +        return NO;
> +    } else {
> +        self.userAccount = [[NXOAuth2AccountStore sharedStore] accounts][accountCount - 1];
Aight, I could throw in a user preference to hold the identifier and get the very last one.
Will do another pull request shortly, let me know what you think about that approach.

—
You are receiving this because you commented.
Reply to this email directly or view it on GitHub